keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,553 @@
+        - name: "PrioritySort" ## QueueSort
+      preScore:
+        disabled:
+        - "*"
+      score:
+        disabled:
+        - "*"
+```
+(Note that in the above, now `PodTopologySpread` will not be registered for `Score` or `PreScore`)
+
+To do this, we must first define a new extension point `MultiPoint` in [`Plugins`](https://github.com/kubernetes/kube-scheduler/blob/bb2d43a/config/v1beta2/types.go#L154-L159)
+```go
+type Plugins struct {
+  MultiPoint PluginSet `json:"multiPoint,omitEmpty"`
+...
+}
+```
+
+We define the `MultiPoint` extension as transparently applicable to all plugins, using 
+type casting to automatically initialize every plugin for every extension point they implement 
+by checking each one. The benefit to this approach is that all plugins are capable of being registered 
+this way without any additional plugin code. The drawbacks of this are less defined control for 
+developers and minor changes to config semantics for users.
+
+Modify [`f.pluginsNeeded`](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L1164) 
+to additionally parse MultiPoint plugins so the function 
+properly builds the map of `pluginName->New function`. Otherwise, we will not have a 
+reference to any plugins that are only enabled through MultiPoint later when we 
+try to unroll them to their individual extension points.
+
+Either before or after [regular extension point plugin initialization](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L349-L353) attempt 
+to initialize all `MultiPoint` plugins as all extension points (this is the section of 
+code that utilizes the factory function stored in the `frameworkImpl` struct)
+
+This code is similar to the existing [`updatePluginList`](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L418)
+function except for 2 notable differences:
+
+1. It does not return an error if a plugin doesn't implement an extension point, it just continues
+2. It is only looping through the MultiPoint config field for each internal extension point
+
+With this implementation, there is no option for plugins or users to define a subset of extension 
+points to enable or disable should we deprecate the existing regular extension points. This is because 
+the plugins themselves do not have control over when their functions are called, and we have cast them 
+into each extension point. At that point the framework, not the plugin, takes over.
+
+Disabling of individual subsets could be preserved using modified config semantics as below:
+
+|   |   |enabled: xyz|disabled: xyz|enabled: *|disabled: *|
+|---|---|------------|-------------|----------|-----------|
+|default|regular|reorder plugins/set weight|disable plugin xyz|no-op|disable all plugins of the point|\
```

Comment:
To make the wording consistent:

disable plugin xyz -> disable plugin xyz for this extension point